src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvThread.cs

```diff
@@ -264,6 +264,19 @@ private void ThreadStart(object parameter)
                 _post.Reference();
                 _post.Dispose();
 
+                // We need this walk because we call ReadStop on 2 places:
+                // 1. On the dispatch pipe UvPipeHandle after reading the correct message
```

Comment:
Is the dispatch pipe now always disposed before this point now that you fixed ListenerPrimary?